bnd.bnd

```diff
@@ -0,0 +1,55 @@
+#
+# MIT License
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+##
+
+# Create OSGi and JPMS module names based on the `groupId` and `artifactId`.
+# This almost agrees with `maven-bundle-plugin`, but replaces non-alphanumeric characters
+# with full stops `.`.
+Bundle-SymbolicName: com.github.packageurl
+-jpms-module-info: $[Bundle-SymbolicName];access=0
+
+# Convert API leakage warnings to errors
+-fixupmessages.priv_refs: "private references";restrict:=warning;is:=error
+
+
+
+# Options specific to dependency packages:
+#
+# Jakarta Validation is an optional dependency.
+Import-Package: \
+  jakarta.validation;resolution:=optional,\
```

Comment:
See PR #204. I think jspecify should also be here as optional.

Reply:
If it's not used as runtime (but is that the case?), change the maven scope to `provided` and set `optional=true`, too, and the OSGi scope could be `requires static`?

Reply:
Yes, if #204 is merged:

- `Import-Package` must have an additional `org.jspecify;resolution:=optional`,
- `-jpms-module-info-options` must have an additional `org.jspecify;transitive=false`, otherwise we end up with a strange `requires static transitive org.jspecify`. Due to the way modules are enumerated `transitive` dependencies are **required** at compile time, even if `static` dependencies are **optional** at runtime.

Reply:
jspecify shouldn't be `provided` - but I'm 99% positive it should be optional. It is only used at compile time by annotation processors.

Reply:
Fixed in https://github.com/package-url/packageurl-java/pull/201/commits/7e988dfcfab42b8b561b04eb9b53e2c8f3abcc5b